psi4/driver/procrouting/scf_proc/scf_iterator.py

```diff
@@ -520,6 +524,10 @@ def scf_iterate(self, e_conv=None, d_conv=None):
                 # make note of the change to early screening; next SCF iteration will be the last
                 early_screening_disabled = True
 
+                # cosx uses the largest grid for its final SCF iteration
+                if cosx_enabled:
+                    self.jk().set_COSX_grid("Final")
```

Comment:
note that this only runs `if early_screening`!

Reply:
Yes, that is correct. Decoupling `early_screening` from direct COSX grid control was one motivation of this PR, to allow for an eventual three-grid COSX SCF scheme.  

Though you make the fair point that `early_screening` still _does_ influence the COSX grid. Maybe I should make some changes to just... get rid of `early_screening` entirely. I wasn't sure what to do with it, honestly, as I mentioned in the Questions section of my PR submission.